src/connector/src/sink/iceberg/compaction.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::{Duration, Instant, SystemTime};
+
+use anyhow::{anyhow, Context};
+use aws_credential_types::provider::SharedCredentialsProvider;
+use aws_sdk_emrserverless::types::builders::SparkSubmitBuilder;
+use aws_sdk_emrserverless::types::{JobDriver, JobRunState};
+use aws_sdk_emrserverless::Client;
+use aws_smithy_runtime_api::client::behavior_version::BehaviorVersion;
+use aws_types::region::Region;
+use tokio::time::sleep;
+use tracing::info;
+
+pub struct NimtableCompactionConfig {
+    region: String,
+    access_key: String,
+    secret_key: String,
+    execution_role_arn: String,
+    application_id: String,
+    entrypoint: String,
+}
+
+impl NimtableCompactionConfig {
+    pub fn from_env() -> Self {
+        NimtableCompactionConfig {
+            region: std::env::var("NIMTABLE_COMPACTION_REGION").unwrap_or_default(),
+            access_key: std::env::var("NIMTABLE_COMPACTION_ACCESS_KEY").unwrap_or_default(),
+            secret_key: std::env::var("NIMTABLE_COMPACTION_SECRET_KEY").unwrap_or_default(),
+            execution_role_arn: std::env::var("NIMTABLE_COMPACTION_EXECUTION_ROLE_ARN")
+                .unwrap_or_default(),
+            application_id: std::env::var("NIMTABLE_COMPACTION_APPLICATION_ID").unwrap_or_default(),
+            entrypoint: std::env::var("NIMTABLE_COMPACTION_ENTRYPOINT").unwrap_or_default(),
+        }
+    }
+}
+
+pub struct NimtableCompactionClient {
+    client: Client,
+    config: NimtableCompactionConfig,
+}
+
+impl NimtableCompactionClient {
+    pub fn new(config: NimtableCompactionConfig) -> Self {
+        let sdk_config = aws_types::SdkConfig::builder()
+            .behavior_version(BehaviorVersion::v2024_03_28())
+            .region(Some(Region::new(config.region.clone())))
+            .credentials_provider(SharedCredentialsProvider::new(
+                aws_credential_types::Credentials::from_keys(
+                    config.access_key.clone(),
+                    config.secret_key.clone(),
+                    None,
+                ),
+            ))
+            .build();
```

Comment:
I think we better use `aws_config::load_from_env()` if the customized credential ENVs named `NIMTABLE_xxx` are absent. In cloud deployment, I think the credential should already been set correctly in the AWS ENVs.

Reply:
I have changed to load the ak sk from the conventional env var when our custom env var is absent.

Reply:
`aws_config::load_from_env` will not just check the ENV var but follows the [AWS default credential provider chain](https://docs.rs/aws-config/latest/src/aws_config/default_provider/credentials.rs.html#25) to fetch credentials. For example, we may set `web_identity_token` or fill-in `~/.aws/config`, `~/.aws/credentials` without setting the AK/SK env vars. Therefore, instead of using the aws AK/SK ENV vars to fill-in NIMTABLE ENV vars, I suggest we call `aws_config::load_from_env()` directly when the credentials are not specified with NIMTABLE ENV vars.